src/main/java/edu/harvard/iq/dataverse/api/datadeposit/SwordServiceBean.java

```diff
@@ -186,7 +187,13 @@ public void setDatasetLicenseAndTermsOfUse(DatasetVersion datasetVersionToMutate
             setTermsOfUse(datasetVersionToMutate, dcterms, null);
         } else {
             License licenseToSet = licenseServiceBean.getByNameOrUri(licenseProvided);
-            if (licenseToSet == null) throw new SwordError("Couldn't find an active license with: " + licenseProvided);
+            if (licenseToSet == null) {
+                List<String> licenses = new ArrayList<>();
+                for (License license : licenseServiceBean.listAllActive()) {
+                    licenses.add(license.getName());
+                }
+                throw new SwordError("Couldn't find an active license with: " + licenseProvided + ". Valid licenses: " + licenses);
```

Comment:
I think you also have to check that licenseToSet.isActive()  - if it is non-null but not isActive, you should get this same error.

Reply:
Ah, good point. Fixed in 66eabc0. Thanks.